packages/compass-components/src/components/file-selector.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { type InputHTMLAttributes, useRef } from 'react';
+
+type FileSelectorTriggerProps = {
+  onClick: () => void;
+};
+
+type FileSelectorProps = Omit<
+  InputHTMLAttributes<HTMLInputElement>,
+  'onChange' | 'onSelect' | 'type' | 'style' | 'ref'
+> & {
+  trigger: (props: FileSelectorTriggerProps) => React.ReactElement;
+  onSelect: (files: File[]) => void;
+};
+
+export function FileSelector({
```

Comment:
Not necessarily completely your problem to solve, but the fact that we now expose two file inputs from compass-components is very confusing. In the spirit of trying to make code universal, maybe we should rename the other one to make it clear that it's electron only and mark it as deprecated with a JSDoc annotation?

Reply:
Renaming it makes sense, but why deprecation?

Reply:
done in [ad151ea](https://github.com/mongodb-js/compass/pull/7127/commits/ad151ea9c6e38e6462f46b56c42818815f7e486a). let me know if it looks good, especially the name :)